Gems/ONNX/Code/Tests/SystemComponentFixture.h

```diff
@@ -0,0 +1,223 @@
+/*
```

Comment:
Did you copy and paste this from EMotionFX?

There is no need for this file in the ONNX gem

Reply:
Do you have any suggestions on where I should look to implement a minimal version of this?

Reply:
And yeah, this is just straight out of EMotionFX, but I was under the impression that this was necessary to initialize everything in order for the tests to run.

Reply:
Yeah a good example would be the WhiteBoxPhysicsTests [EditorWhiteBoxPhysicsTestEnvironment](https://github.com/o3de/o3de/blob/development/Gems/WhiteBox/Code/Tests/WhiteBoxPhysicsTest.cpp#L27-L58) class.
You can also look at the PhysXEditorTest [PhysXEditorTestEnvironment](https://github.com/o3de/o3de/blob/development/Gems/PhysX/Code/Tests/PhysXEditorTest.cpp#L46-L125) class as a backup.

Afterwards at the end of your ONNXEditorTest.cpp you would replace the line of 
```c++
AZ_UNIT_TEST_HOOK(DEFAULT_UNIT_TEST_ENV);
```

With 
```cpp
AZ_UNIT_TEST_HOOK(new <NameOfTheGemEnvironmentClassYouCreated>);
```

Which adds the `AzRunUnitTest` extern "C" function that the AzTestRunner picks on to run googletest

Reply:
Changed